operator/main.go

```diff
@@ -63,16 +66,16 @@ func main() {
 	logger := zap.New(zap.UseFlagOptions(&opts))
 	ctrl.SetLogger(logger)
 
-	watchNamespace := namespace
+	watchNamespaceConfig := map[string]cache.Config{namespace: {}}
```

Comment:
description of the new api is [here](https://github.com/kubernetes-sigs/controller-runtime/blob/9bc967a711e9538cff4099645dd6df64c9ba8a74/pkg/cache/cache.go#L181-L192)